game_patch/misc/vpackfile.cpp

```diff
@@ -291,27 +340,62 @@ static int vpackfile_build_file_list_new(const char* ext_filter, char*& filename
     if (g_is_overriding_disabled) {
         // Don't allow overriding files after game is initialized because it can lead to crashes
         return false;
     }
-    if (!new_entry->parent->is_user_maps) {
-        // Allow overriding by packfiles from game root and from mods
+    // Allow overriding by packfiles from game root and from mods (not user_maps or client_mods)
+    else if (!new_entry->parent->is_user_maps && !new_entry->parent->is_client_mods) {
         return true;
     }
-    if (!old_entry->parent->is_user_maps && !stricmp(rf::file_get_ext(new_entry->name), ".tbl")) {
-        // Always skip overriding tbl files from game by user_maps
-        return false;
-    }
 #ifdef MOD_FILE_WHITELIST
-    if (is_mod_file_in_whitelist(new_entry->file_name)) {
+    else if (is_mod_file_in_whitelist(new_entry->name)) {
         // Always allow overriding for specific files
         return true;
     }
 #endif
-    if (!g_game_config.allow_overwrite_game_files) {
-        return false;
+    // Process if in client_mods
+    else if (new_entry->parent->is_client_mods) {
+        if (is_tbl_file_in_allowlist(new_entry->name) || is_tbl_file_a_hud_messages_file(new_entry->name)) {
+            // allow overriding of tbl files from client_mods if they are on allow list or are _text.tbl
+            g_is_modded_game = true; // goober todo: confirm what this is used for. If anticheat, can remove from here
+            return true;
+        }
+        else if (is_tbl_file(new_entry->name)) {
+            // deny all other tbl overrides
+            return false;
+        }
+        g_is_modded_game = true;
+        return true;
+    }
+    // Process if in user_maps
+    else if (new_entry->parent->is_user_maps) {
+        if (old_entry->parent->is_user_maps) {
+            // allow files from user_maps to override other files in user_maps, even if the switch is off
+            // files from user_maps can't override files from client_mods because client_mods is parsed later
+            return true;
+        }
+        else if (!g_game_config.allow_overwrite_game_files)
+        {
+            // if the switch is off, no overwriting with files from user_maps, except for other files in user_maps
+            return false;
+        }
+        else if (is_tbl_file_in_allowlist(new_entry->name) || is_tbl_file_a_hud_messages_file(new_entry->name))
+        {
+            // allow overriding of tbl files from user_maps if they are on allow list or are _text.tbl
+            g_is_modded_game = true; // goober todo: confirm what this is used for. If anticheat, can remove from here
+            return true;
+        }
+        else if (is_tbl_file(new_entry->name))
+        {
+            // deny all other tbl overrides
+            return false;
+        }
+        g_is_modded_game = true;
+        return true;
     }
+    // resolve warning by having a default option, even though there should be no way to hit it
```

Comment:
to be honest this function got very complicated. Not sure if we want to handle so many corner cases distinctly. Why do we even handle tbl mods in `user_maps` if we add `client_mods`? What's the point?